src/mesido/esdl/esdl_additional_vars_mixin.py

```diff
@@ -16,8 +16,100 @@
 class ESDLAdditionalVarsMixin(CollocatedIntegratedOptimizationProblem):
     __temperature_options = {}
 
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
     def read(self):
         super().read()
+        parameters = self.parameters(0)
+        bounds = self.bounds()
+
+        # Here we do a check between the available pipe classes and the demand profiles. This is to
+        # ensure that we don't have unneeded large amount of available pipe classes for pipes
+        # connected to smaller demands.
+        # TODO: add the same for electricity ones we have proper support for that in the ESDLMixin
+        for asset, (connected_asset, _orientation) in self.energy_system_topology.demands.items():
+            if asset in self.energy_system_components.get("gas_demand", []):
+                try:
+                    max_demand = min(
+                        max(self.get_timeseries(f"{asset}.target_gas_demand").values),
+                        bounds[f"{asset}.Gas_demand_mass_flow"][1],
+                    )
+                except KeyError:
+                    max_demand = bounds[f"{asset}.Gas_demand_mass_flow"][1]
+                new_pcs = []
+                found_pc_large_enough = False
+                for pc in self.gas_pipe_classes(connected_asset):
+                    if not found_pc_large_enough:
+                        new_pcs.append(pc)
+                        if new_pcs[-1].maximum_discharge >= max_demand:
+                            found_pc_large_enough = True
+                self._override_gas_pipe_classes[connected_asset] = new_pcs
+            if asset in self.energy_system_components.get("heat_demand", []):
+                try:
+                    max_demand = min(
+                        max(self.get_timeseries(f"{asset}.target_heat_demand").values),
+                        bounds[f"{asset}.Heat_demand"][1],
+                    )
+                except KeyError:
+                    max_demand = bounds[f"{asset}.Heat_demand"][1]
+                new_pcs = []
+                found_pc_large_enough = False
+                for pc in self.pipe_classes(connected_asset):
```

Comment:
We should not have the DN0 option for the connected pipe of a demand if there was a target_demand

Reply:
Check if this maybe goes wrong with logical links, as the connected asset might not be a pipe and hnce has no pipe classes

Reply:
only add pipe to new pcs that has sufficient capacity for the max demand. This means that at most 2 pcs will be added, the one which is sufficient for the peak and optionally one which is suffcient for the peak +30%. Note that there can be cases that one pc suffices for both and only one pc would be added.

Reply:
the pipe_classes() method will return an empty list in case of a logical link and hence the loop would not be entered. Thus it's no problem in case of a logical link

Reply:
DN0 zero is now removed for heat_demand